File: Beginner/Farmer And His Plot (RECTSQ)/farmer.c
#include<stdio.h>


int gcd(int A,int B) {
    if(B==0)
    return A;
    else
    return gcd(B,A%B);
}

int main() {
    int t, A, B, c; 
    scanf("%d",&t);

    while(t--) {
        scanf("%d%d",&A,&B);
        c = gcd(A,B);
        printf("%d\n",(A*B)/(c*c));
    }
    return 0;
}

File: Beginner/Farmer And His Plot (RECTSQ)/farmer.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner input = new Scanner(System.in);
		
		int t = input.nextInt();
		
		for(int i=0; i<t; i++)
		{
		    int length = input.nextInt();
		    int breadth = input.nextInt();
		    
		    int area = length*breadth;
		    int gcd=1;
		    
		    for(int val = 1; val <= length && val <= breadth; val++)
            {
                if(length%val==0 && breadth%val==0)
                {
                    gcd = val;
                }
            }
            
            int result = (int) (area/(gcd*gcd));
            System.out.println(result);
		}
	}
}
